### PR TITLE
Add quickstart guide for DHCP

### PR DESCRIPTION
**Pull request by mathewab**
Added quickstart guide for DHCP
Made minor corrections to other documentation

### CODE REVIEW

File: templates/guides/quickstart-dhcp.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+  }
+}
+
+provider "bloxone" {
+  csp_url = "https://csp.infoblox.com"
+  api_key = "<BloxOne API Key>"
+}
+````
+
+!> Warning: Hard-coded credentials are not recommended in any configuration file. It is recommended to use environment variables.
```
Comment (swoirgard) [inline templates/guides/quickstart-dhcp.md:34]: is that a valid markdown ?

File: templates/guides/quickstart-dhcp.md [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+// Create a subnet within the address block
+resource "bloxone_ipam_subnet" "this" {
+  space   = bloxone_ipam_ip_space.this.id
+  address = "10.0.0.0"
+  cidr    = "24"
+}
+
+````
+
+We can now run `terraform plan` to see what resources will be created.
```
Comment (swoirgard) [inline templates/guides/quickstart-dhcp.md:78]: Guides are usually written in an impersonal style. The beginning of the guid talks to the user with "You", it should continue to do so all along.

File: templates/guides/quickstart-dhcp.md [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+
+```shell
+terraform plan
+```
+
+Further, we will create a range and a fixed address reservation within the subnet.
+
+We will use the following resources to create these
```
Comment (swoirgard) [inline templates/guides/quickstart-dhcp.md:86]: "You"

File: templates/guides/quickstart-dhcp.md [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+
+// Create a range within the subnet
+resource "bloxone_ipam_range" "this" {
+  space = bloxone_ipam_ip_space.this.id
+  start = "10.0.0.5"
+  end   = "10.0.0.100"
+  dhcp_options = [
+    {
+      option_code  = local.dhcp4_option_code_lookup["domain-name-servers"]
+      option_value = "1.1.1.1"
```
Comment (swoirgard) [inline templates/guides/quickstart-dhcp.md:125]: Let's avoid promoting a competitor (cloudflare's dns address)...
Use an address within the 10/16 network.

File: templates/guides/quickstart-dhcp.md [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+
+````terraform
+provider "aws" {
+  region     = "us-west-2"
+  access_key = "my-access-key"
+  secret_key = "my-secret-key"
+}
+```` 
+
+!> Warning: Hard-coded credentials are not recommended in any configuration file. It is recommended to use the AWS credentials file or environment variables.
```
Comment (swoirgard) [inline templates/guides/quickstart-dhcp.md:179]: !> markdown ?

File: templates/guides/quickstart-dhcp.md (hunk 1 above)
Comment (mathewab) [reply templates/guides/quickstart-dhcp.md:34]: Terraform Registry documentation accepts it as a special case. It shows up as a colored callout box. 

https://developer.hashicorp.com/terraform/registry/providers/docs#callouts

File: Makefile [hunk 6]
```diff
@@ -24,5 +24,6 @@ testacc:
 
 gen:
 	go generate
+	terraform-docs ./modules/bloxone_infra_host_aws
```
[the remark this answers is not in the document]
Comment (mathewab) [reply Makefile:27]: This doesn't work. The `terraform-docs` requires the proper directory and does not recursively figure out the sub directories

[the remark this answers is not in the document]
Comment (venkat-iblox) [reply Makefile:27]: https://terraform-docs.io/user-guide/configuration/recursive/
Can we try this?

[the remark this answers is not in the document]
Comment (mathewab) [reply Makefile:27]: Tried it, still doesn't work correctly. That works only for child modules. Our directory structure is a bit different.

File: templates/guides/quickstart-dhcp.md (hunk 4 above)
Comment (mathewab) [reply templates/guides/quickstart-dhcp.md:127]: Changed to 10.0.0.1

File: templates/guides/quickstart-dhcp.md (hunk 2 above)
Comment (mathewab) [reply templates/guides/quickstart-dhcp.md]: Fixed the "we" language.

**Review by swoirgard — APPROVED** [review]

**Review by swoirgard — APPROVED** [review]

**Review by venkat-iblox — APPROVED** [review]

**Pull request merged**